core/src/cluster_info.rs

```diff
@@ -1040,7 +1036,21 @@ impl ClusterInfo {
                     if exit.load(Ordering::Relaxed) {
                         return;
                     }
-                    obj.write().unwrap().purge(timestamp());
+                    let timeout = {
+                        if let Some(ref bank_forks) = bank_forks {
+                            let bank = bank_forks.read().unwrap().working_bank();
+                            let epoch = bank.epoch();
+                            let epoch_schedule = bank.epoch_schedule();
+                            let ms_per_slot = (1.0f64 / DEFAULT_TICKS_PER_SECOND as f64
+                                * DEFAULT_TICKS_PER_SLOT as f64
```

Comment:
Let's make ms_per_slot a constant in timing.rs? Also we can do this without floating point math. 1000*ticks per slot and then divide by ticks per second.